### PR TITLE
feature: target api to netstandard1.1, netstandard2.0 and net45

### PR DESCRIPTION
**Pull request by marcovisserFurore**
This PR replaces PR #639 . 
Thanks @IvanFarkas for your initial pull request. We could not use your pull request, because we targeting other frameworks. But we used your PR for inspiration.

In this PR the FHIR .NET Api targets netstandard1.1, netstandard2.0 and net45. The unittests are running now under netcoreapp2.1 (should be faster).

### CODE REVIEW

File: src/Hl7.Fhir.Specification.Tests/Source/SummarySourceTests.cs [hunk 1]
```diff
@@ -147,11 +147,7 @@ public async Tasks.Task TestThreadSafety()
                 tasks[i] = Tasks.Task.Run(
                     () =>
                     {
-#if DOTNETFW
                         var threadId = System.Threading.Thread.CurrentThread.ManagedThreadId;
```
Comment (ewoutkramer) [inline src/Hl7.Fhir.Specification.Tests/Source/SummarySourceTests.cs]: Idem

File: src/Hl7.Fhir.Specification/Specification/Source/DirectorySource.cs [hunk 2]
```diff
@@ -42,14 +40,9 @@ namespace Hl7.Fhir.Specification.Source
     [DebuggerDisplay(@"\{{DebuggerDisplay,nq}}")]
     public class DirectorySource : ISummarySource, IConformanceSource, IArtifactSource
     {
-        // netstandard has no CurrentCultureIgnoreCase comparer
-#if DOTNETFW
         private static readonly StringComparer ExtensionComparer = StringComparer.InvariantCultureIgnoreCase;
```
Comment (ewoutkramer) [inline src/Hl7.Fhir.Specification/Specification/Source/DirectorySource.cs]: Waarom is de #if hier weg? Er was blijkbaar een verschil tussen InvariantCultureIgnoreCase en OrdinalCultureIgnoreCase, dat valt nu weg.

File: src/Hl7.Fhir.Specification/Specification/Source/DirectorySource.cs [hunk 3]
```diff
@@ -9,8 +9,6 @@
 // [WMR 20171023] TODO
 // - Allow configuration of duplicate canonical url handling strategy
 
-#if NET_FILESYSTEM
```
Comment (ewoutkramer) [inline src/Hl7.Fhir.Specification/Specification/Source/DirectorySource.cs]: Hoe kan deze weg zijn?  Onder NETSTD11 hoort het dan niet meer te compileren, toch?

File: src/Hl7.Fhir.Specification/Specification/Source/Summary/ArtifactSummaryExtensions.cs [hunk 4]
```diff
@@ -6,8 +6,6 @@
  * available at https://github.com/ewoutkramer/fhir-net-api/blob/master/LICENSE
  */
 
-#if NET_FILESYSTEM
```
Comment (ewoutkramer) [inline src/Hl7.Fhir.Specification/Specification/Source/Summary/ArtifactSummaryExtensions.cs]: Je zou verwachten dat er filesystem code in de unit staat, en dat het niet compileert onder NETSTD11, toch?

**Comment by IvanFarkas** [comment]
That is awesome. I glad I was able to help. Great lob!

**Comment by IvanFarkas** [comment]
Does this make me a contributor?

File: src/Hl7.Fhir.Specification.Tests/Source/ConformanceSourceTests.cs [hunk 5]
```diff
@@ -394,11 +394,7 @@ public async Task TestThreadSafety()
                 tasks[i] = Task.Run(
                     () =>
                     {
-#if DOTNETFW
                         var threadId = System.Threading.Thread.CurrentThread.ManagedThreadId;
```
[the remark this answers is not in the document]
Comment (marcovisserFurore) [reply src/Hl7.Fhir.Specification.Tests/Source/ConformanceSourceTests.cs]: Let op: dit is een unittest en draait nu op netcoreapp2.1. En daar zit dit zeker in

File: src/Hl7.Fhir.Specification.Tests/Source/SummarySourceTests.cs (hunk 1 above)
Comment (marcovisserFurore) [reply src/Hl7.Fhir.Specification.Tests/Source/SummarySourceTests.cs]: Let op: dit is een unittest en draait nu op netcoreapp2.1. En daar zit dit zeker in

File: src/Hl7.Fhir.Specification/Specification/Source/Summary/ArtifactSummaryExtensions.cs (hunk 4 above)
Comment (marcovisserFurore) [reply src/Hl7.Fhir.Specification/Specification/Source/Summary/ArtifactSummaryExtensions.cs]: HL7.Fhir.Specification targets only net45 and netstandard2.0. So no netstandard1.1 anymore. That's why this conditional directive could be deleted.

File: src/Hl7.Fhir.Specification/Specification/Source/DirectorySourceSettings.cs [hunk 6]
```diff
@@ -81,21 +77,15 @@ public void CopyTo(DirectorySourceSettings other)
...
         /// <summary>Creates a new <see cref="DirectorySourceSettings"/> object that is a copy of the current instance.</summary>
         object ICloneable.Clone() => Clone();
-#endif
 
         /// <summary>Returns the default content directory of the <see cref="DirectorySource"/>.</summary>
         public static string SpecificationDirectory
         {
             get
             {
-#if DOTNETFW
```
[the remark this answers is not in the document]
Comment (marcovisserFurore) [reply src/Hl7.Fhir.Specification/Specification/Source/DirectorySourceSettings.cs]: HL7.Fhir.Specification targets only net45 and netstandard2.0. So no netstandard1.1 anymore. That's why this conditional directive could be deleted.

File: src/Hl7.Fhir.Specification/Specification/Source/DirectorySource.cs (hunk 3 above)
Comment (marcovisserFurore) [reply src/Hl7.Fhir.Specification/Specification/Source/DirectorySource.cs]: HL7.Fhir.Specification targets only net45 and netstandard2.0. So no netstandard1.1 anymore. That's why this conditional directive could be deleted.

File: src/Hl7.Fhir.Specification/Specification/Source/DirectorySource.cs (hunk 2 above)
Comment (marcovisserFurore) [reply src/Hl7.Fhir.Specification/Specification/Source/DirectorySource.cs]: HL7.Fhir.Specification targets only net45 and netstandard2.0. So no netstandard1.1 anymore. That's why this conditional directive could be deleted.

**Comment by ewoutkramer** [comment]
> Does this make me a contributor?

You can do a PR on the contributors file ;-)

File: src/Hl7.Fhir.Core/Rest/WebRequestExtensions.cs [hunk 7]
```diff
@@ -21,7 +21,7 @@ public static class WebRequestExtensions
     {
         internal static void WriteBody(this HttpWebRequest request, bool CompressRequestBody, byte[] data)
         {
-#if !DOTNETFW
+#if NETSTANDARD1_1
```
[the remark this answers is not in the document]
Comment (ewoutkramer) [reply src/Hl7.Fhir.Core/Rest/WebRequestExtensions.cs]: Ok, (sorry, switching back to English ;-), this means we don't have compression under 1.1  - so we should make sure the properties to control compression are not available on the client in that case!

File: src/Hl7.Fhir.Specification.Tests/Source/ConformanceSourceTests.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (ewoutkramer) [reply src/Hl7.Fhir.Specification.Tests/Source/ConformanceSourceTests.cs]: Oh, ja ;-)

File: src/Hl7.Fhir.Specification/Specification/Source/DirectorySource.cs (hunk 3 above)
Comment (ewoutkramer) [reply src/Hl7.Fhir.Specification/Specification/Source/DirectorySource.cs]: Oh crap, yeah, forgot about that.

**Comment by ewoutkramer** [comment]
@marcovisserFurore  - I think the result of our conclusion is that
* We want to make sure no compression-related properties appear on the Client under NET1.1
* I'd like to review the compression code in the FHIRClient since I think it either can be made simpler (much of the current Waits removed) - that's a separate PR.
* Unfortunately, this PR no longer merges cleanly due to other PRs. Can you fix that?

File: src/Hl7.Fhir.Core/Rest/WebRequestExtensions.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (marcovisserFurore) [reply src/Hl7.Fhir.Core/Rest/WebRequestExtensions.cs]: I will remove the compression properties on FhirClient for NETSTD1.1

**Comment by marcovisserFurore** [comment]
@ewoutkramer - I already have the compression-related properties removed under netstandard1.1 (this can cause breaking change for our customers). The commit I'll do tomorrow.
I will update the PR, so there will be no conflicts

**Pull request merged**